src/delegate/index.ios.ts

```diff
@@ -17,35 +12,40 @@ function setup() {
     }
 
     private static getAppDelegate() {
+      // As of NativeScript 8.2, ensureNativeApplication should be called prior to accesses to applicationModule.ios.
+      /* @ts-ignore */
+      if (!!applicationModule.ensureNativeApplication) {
+        /* @ts-ignore */
+        applicationModule.ensureNativeApplication();
+      }
       // Play nice with other plugins by not completely ignoring anything already added to the appdelegate
       if (applicationModule.ios.delegate === undefined) {
-        @ObjCClass(UIApplicationDelegate)
-        class UIApplicationDelegateImpl extends UIResponder implements UIApplicationDelegate { }
-  
+        @NativeClass
+        class UIApplicationDelegateImpl extends UIResponder implements UIApplicationDelegate {
+          public static ObjCProtocols = [UIApplicationDelegate];
+        }
+
         applicationModule.ios.delegate = UIApplicationDelegateImpl;
       }
       return applicationModule.ios.delegate;
     }
-  
+
     private static addAppDelegateMethods = appDelegate => {
-      if (parseInt(platformModule.Device.osVersion.split('.')[0]) >= 10 ) {
-        // iOS >= 10
-        appDelegate.prototype.applicationOpenURLOptions = (
-              application: UIApplication,
-              url: NSURL,
-              options: NSDictionary<string, any>) => {
-          TnsOAuthClientAppDelegate.handleIncomingUrl(url);
-        };
-      } else {
-        // iOS < 10
-        appDelegate.prototype.applicationOpenURLSourceApplicationAnnotation = (
-              application: UIApplication,
-              url: NSURL,
-              sourceApplication: string,
-              annotation: any ) => {
-          TnsOAuthClientAppDelegate.handleIncomingUrl(url);
-        };
-      }
+      // iOS >= 10
+      appDelegate.prototype.applicationOpenURLOptions = (
+        application: UIApplication,
+        url: NSURL,
+        options: NSDictionary<string, any>) => {
+        TnsOAuthClientAppDelegate.handleIncomingUrl(url);
+      };
+      // iOS < 10
+      appDelegate.prototype.applicationOpenURLSourceApplicationAnnotation = (
+        application: UIApplication,
+        url: NSURL,
+        sourceApplication: string,
+        annotation: any) => {
+        TnsOAuthClientAppDelegate.handleIncomingUrl(url);
+      };
```

Comment:
Is this safe to do? I mean it's JS, if the property does not exist on the prototype it should work just fine, but I can't help but think the iOS version check was there for a reason.

I want to say "It might be worthwhile to test on iOS < 10", but on the other hand the [market share for iOS 9 and before is not even on the chart going back to 2017](https://gs.statcounter.com/ios-version-market-share/mobile/worldwide/#monthly-201706-202206) so dropping support entirely is fine IMO.